src/app/options.cpp

```diff
@@ -235,6 +240,68 @@ namespace
         return o == s;
     }
 
+    // Option that is explicitly set to true or false, and whose value is undefined when unspecified.
+    // May not have a shortcut.
+    class TriStateBoolOption: protected StringOption
+    {
+    public:
+        constexpr TriStateBoolOption(const char *name)
+            : StringOption {name}
+        {
+        }
+
+        using StringOption::operator==;
+        
+        QString usage() const
+        {
+            return StringOption::usage(QLatin1String("true|false"));
+        }
+
+        TriStateBool value(const QString &arg) const
+        {
+            QString val = StringOption::value(arg);
+            
+            if (val.toUpper() == QLatin1String("TRUE") || val == QLatin1String("1")) {
+                return TriStateBool::True;
+            }
+            else if (val.toUpper() == QLatin1String("FALSE") || val == QLatin1String("0")) {
+                return TriStateBool::False;
+            }
+            else {
+                throw CommandLineParameterError(QObject::tr("Parameter '%1' must follow syntax '%1=%2'",
+                                                            "e.g. Parameter '--add-paused' must follow syntax "
+                                                            "'--add-paused=<true|false>'")
+                                                .arg(fullParameter())
+                                                .arg(QLatin1String("<true|false>")));
+            }
+        }
+
+        TriStateBool value(const QProcessEnvironment &env) const
+        {
+            QString val = env.value(envVarName());
+            
+            if (val.isEmpty()) {
+                return TriStateBool::Undefined;
+            }
+            else if (val.toUpper() == QLatin1String("TRUE") || val == QLatin1String("1")) {
+                return TriStateBool::True;
+            }
+            else if (val.toUpper() == QLatin1String("FALSE") || val == QLatin1String("0")) {
+                return TriStateBool::False;
+            }
+            else {
+                qDebug() << QObject::tr("Expected %1 in environment variable '%2', but got '%3'")
+                    .arg(QLatin1String("true|false")).arg(envVarName()).arg(val);
+                return TriStateBool::Undefined;
+            }
+        }
+    };
+
+    inline bool operator==(const QString &s, const TriStateBoolOption &o)
```

Comment:
if it's not critical, please remove `inline`, I prefer to trust the compiler decision.